app/javascript/image_sharing/components/test/FeedbackForm.spec.js

```
@@ -1,15 +1,38 @@
-import React from 'react';
-import { mount } from 'enzyme';
 import chai, { expect } from 'chai';
-import { describe, it } from 'mocha';
 import chaiEnzyme from 'chai-enzyme';
+import { mount } from 'enzyme';
+import { afterEach, beforeEach, describe, it } from 'mocha';
+import React from 'react';
+import sinon from 'sinon';
 
 import FeedbackForm from '../FeedbackForm';
+import { FeedbackStore } from '../../stores/FeedbackStore';
 
 chai.use(chaiEnzyme());
 
+const stubActionOnStore = (store, actionProp) => {
+  const stub = sinon.stub();
+  Object.defineProperty(
+    store,
+    actionProp,
+    { configurable: true, writeable: true, enumerable: false, value: stub }
```

I've never had to do this, but I think I've seen this is a solution to some of my issues on google; did you have to add this to get around something mobx was not allowing you to stub?

---

Yes. I am not sure it applies to all properties or just those with annotations, but without this, stubbing just silently fails and the real code is still called.

---

When we run into this in prod we can look around a bit; I think there is a way to stub the decorated/annotated store methods and have it work, but it is very easy to do things in a certain order such that the store's real code is called